facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -348,57 +388,55 @@ private static int getNotFoundArtifactRating(org.jboss.pnc.model.Artifact artifa
-            Set<Base32LongID> pncBuiltRecordIds,
             User user) {
 
         // The artifact was not built from source, but could already be present as a dependency recorded in PNC system.
-        // To avoid unnecessary artifact duplication (see NCLSUP-990), we will search for a best matching artifact with
-        // some priority checks.
-
-        List<org.jboss.pnc.model.Artifact> artifacts;
-        Optional<org.jboss.pnc.model.Artifact> bestMatch;
-
-        // 1) We will see if there are artifacts with the same SHA-256 which are dependencies of one of the found
-        // PNC builds (if any) in the current delivered analysis.
-        // If more than one artifact is found, find a best match.
-        if (!pncBuiltRecordIds.isEmpty()) {
-            artifacts = artifactRepository
-                    .withSha256AndDependantBuildRecordIdIn(artifact.getSha256(), pncBuiltRecordIds);
-            bestMatch = getBestMatchingArtifact(artifacts);
-            if (bestMatch.isPresent()) {
-                return bestMatch.get();
-            }
-        }
-
-        // 2) We will see if there is an artifact with the same SHA-256 and identifier.
-        // If more than one artifact is found, find a best match.
-        if (artifact.getArtifactType() == ArtifactType.MAVEN) {
-            String identifier = createIdentifier((MavenArtifact) artifact);
-            artifacts = artifactRepository.withIdentifierAndSha256(identifier, artifact.getSha256());
-            bestMatch = getBestMatchingArtifact(artifacts);
-            if (bestMatch.isPresent()) {
-                return bestMatch.get();
-            }
-        } else {
-            artifacts = artifactRepository.withIdentifierAndSha256(artifact.getFilename(), artifact.getSha256());
-            bestMatch = getBestMatchingArtifact(artifacts);
-            if (bestMatch.isPresent()) {
-                return bestMatch.get();
-            }
-        }
+        // To avoid unnecessary artifact duplication (see NCLSUP-990), we will search for a best matching artifact.
+        // PLEASE NOTE that we don't know much about such artifacts, for example whether they are Maven based. We really
+        // know just their SHA and their filename. Their identifier is their filename.
+        Path path = Paths.get(artifact.getFilename());
+        String filename = path.getFileName().toString();
 
-        // 3) We will see if there is an artifact just with the same SHA-256.
-        // If more than one artifact is found, find a best match.
-        artifacts = artifactRepository.withSha256In(Collections.singleton(artifact.getSha256()));
-        bestMatch = getBestMatchingArtifact(artifacts);
-        if (bestMatch.isPresent()) {
-            return bestMatch.get();
+        // Search for artifacts with the same SHA-256 and filter for its name. If no matches are found, create the
+        // artifact. Yes, if an artifact was renamed in the ZIP, we will create a new entry in the DB.
```

Comment:
Because I realized that in case of not found artifacts, we know very few things about them from the BF report : just the checksums and the filename. The "identifier" was also not available because the local archives from BF were not Maven artifacts (BF does not know anything about them, and sets their identifier to the filename). Basically the search for identifier and sha256 was never done for the not found artifacts.
So, the only other thing we could rely beside the sha256 is the filename. In case of same sha256 with different identifiers (and thus, filenames), we can filter by the filename and see if there is a match. If no match is found (or multiple matches are found), the safest thing is to actually create a new artifact in the DB. I saw cases where a ".jar", a "-sources.jar" and "-javadoc.jar" had the same sha256... I realize this is a regression from the last improvement made, but it could made very wrong assumptions . Considering the other improvements we have done, I believe this is a good compromise?

Reply:
ok, makes sense